### ISSUE TITLE
Permalink anchor UI

### ISSUE THREAD

**Issue by wking** [jbenet/depviz#45]
The “navigate by clicking on nodes” idea has come up a few times (#2, #17, #30), but the current implementation is based on dragging the `viewBox` around.  As I [mentioned][1] in #17, cards are a bit crowded to make the background that click-to-center target.  #30 proposed using the issue title, but I don't think that's particularly intuitive.  Stealing a bit of space by following the dep-indicator pattern with an explicit link symbol gives us something like this:

![permalink](https://cloud.githubusercontent.com/assets/209920/20829062/49756b2e-b82f-11e6-9788-20805bfef197.png)

![permalink-collapsed](https://cloud.githubusercontent.com/assets/209920/20829070/4d7788e2-b82f-11e6-83e6-df3b4b91b41a.png)

That starts to look a bit warty, but it is fairly clear.  You could also inline the link:

![permalink-inline](https://cloud.githubusercontent.com/assets/209920/20829193/e4a465b4-b82f-11e6-8f8b-3a41b2f16372.png)

but having it float inside the card makes it feel like issue information, and less like part of the graph UI.

I'm leaning towards the warty lower-left circle, but would be happy to switch to something else if someone has a cleaner idea :).

[1]: https://github.com/jbenet/depviz/issues/17#issuecomment-263368416

**Comment by jbenet** [comment]
- in the cases i've shown depviz to others, people have always wanted to navigate to re-center on a card by clicking on it (background or the title. they actually expect the vicinity, and anywhere on the card to work -- except obvious other things: a user picture, comments' number, etc)
- I (and others) would expect the "link circle" icon to copy a link to depviz centered on that issue to clipboard (but not recenter)
- I (and others) would expect the github icon to link me to the github.com website version of this task. (not `https://github.com`)

I think we should:
- make the whole card a link, to `#<depviz-uri>` (i.e. to navigate to that one)
- make the github icon link to the issue on github
- the title could be either github or recenter. i can see cases for both.
- add the link icon which should appear on-hover only, as a way to copy that task's depviz link to clipboard. (but not to navigate/recenter)
- in all cases, we should tolerate clicking on a small margin around the object. (Eg small author pictures)

**Comment by jbenet** [comment]
(for those with email: updated comment above)

**Comment by Kubuxu** [comment]
Entire cards are draggable which is probably why they are not clickable.

**Comment by wking** [comment]
On Sun, Dec 04, 2016 at 09:29:05PM -0800, Juan Benet wrote:
> - I (and others) would expect the "link circle" icon to copy a link
>   to depviz centered on that issue to clipboard (but not recenter)

Is there an alternative symbol you'd prefer?  Maybe ⚓ (U+2693 ANCHOR)?
📌 (U+1F4CC PUSHPIN)?  # (U+0023 NUMBER SIGN)?

Or is the only option for “center on this card and update the window
location” the card background?  “Hover” is less accessible on touch
UIs, so I'd rather not have too much implicit behavior like “the whole
card does …”.  But we can always document it on the landing page if we
do need to make the whole card do something.

**Comment by VictorBjelkholm** [comment]
@Kubuxu 
> Entire cards are draggable which is probably why they are not clickable.

No, that would be two different actions, look at how Trello handles having both without problems for example.

> Is there an alternative symbol you'd prefer? Maybe ⚓ (U+2693 ANCHOR)? :pushpin: (U+1F4CC PUSHPIN)? # (U+0023 NUMBER SIGN)? 

For copy, there is a establish icon, the holy pasteboard icon, :clipboard: 📋 (U+1F4CB) but would be better to move into proper icons rather than unicode symbols actually. For example: http://www.flaticon.com/search?word=copy

> But we can always document it on the landing page if we do need to make the whole card do something

In general, I think we should have the idea that if something becomes so complex without it's actually being real complex (think rocket science), we should make the features and actions easy enough to be self-describing.

**Comment by jbenet** [comment]
> In general, I think we should have the idea that if something becomes so complex without it's actually being real complex (think rocket science), we should make the features and actions easy enough to be self-describing.

agreed.

though i think clicking/touching through on the card is super obvious. it's the first thing people try.
